integrationtest/integration_test.go

```diff
@@ -18,26 +20,56 @@ import (
 //   - A Kubernetes cluster in which:
 //       - it can use the `test` namespace
 //       - Vault is deployed and accessible
+//       - There is a serviceaccount called test-token-reviewer-account with access to the TokenReview API
```

Comment:
Interesting, I think the default token in the vault-0 pod should already have access to the TokenReivew API: https://github.com/hashicorp/vault-helm/blob/4ae52c8bd3d3f4d303196ecb455ae07d9a943ba8/templates/server-clusterrolebinding.yaml#L17-L25